JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToInteraction/UniqueToInteraction.tsx

```diff
@@ -15,31 +15,36 @@ import SaveIcon from '@mui/icons-material/Save'
 import CancelIcon from '@mui/icons-material/Close'
 import { useEffect, useState } from 'react'
 import { EditToolbar } from 'components/shared/EditToolBar'
+import { useConfiguration } from 'hooks/useUIConfiguration'
+import { Configuration } from 'types/Configuration'
+import { toSnakeCase } from 'utils/helpers'
 
-const UniqueToInteraction = ({
-  uniqueInteractionData
-}: {
-  uniqueInteractionData: any
-}) => {
-  const [rows, setRows] = useState(uniqueInteractionData)
+const UniqueToInteraction = () => {
+  const [rows, setRows] = useState<any[]>([])
   const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({})
+  const { configuration, setConfiguration } = useConfiguration()
 
   useEffect(() => {
-    const rowsWithIds = uniqueInteractionData.map(
-      (row: any, index: number) => ({
-        ...row,
-        id: index.toString()
-      })
-    )
-    setRows(rowsWithIds)
-  }, [uniqueInteractionData])
+    if (configuration && Array.isArray(configuration.auxInteractionFields)) {
+      const rowData = configuration.auxInteractionFields.map(
+        (row: any, rowIndex: number) => ({
+          id: rowIndex + 1,
+          ...row,
+          rowIndex
+        })
+      )
+      setRows(rowData)
+    }
+  }, [configuration])
 
   const handleEditClick = (id: GridRowId) => () => {
     setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.Edit } })
   }
-
   const handleSaveClick = (id: GridRowId) => () => {
-    setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } })
+    const updatedRow = rows.find((row: { id: GridRowId }) => row.id === id)
+    if (updatedRow) {
+      setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } })
```

Comment:
Hi @NyashaMuusha! Just checking what the scenario would be here requiring this second call to "setRowModesModel". Seems like it might be a redundant call?